### ISSUE TITLE
`conditional` computes unnecessary property

### ISSUE THREAD

**Issue by JackCA** [kellyselden/ember-awesome-macros#275]
I am not sure whether this is intended or avoidable behavior, but in a scenario where you have the following:

``` javascript
condition: false,

property: conditional('condition', 'trueProp', 'falseProp'),

falseProp: computed(function() {
  // we expect this to be called
  return 'good'
}),

trueProp: computed(function() {
  // but not this one
  // this property could incorrectly assume that `condition` is true
  // or it could be very slow
  // etc
})
```

Essentially, the second argument to `condition` could possibly not be executed until the first argument becomes true.

Thoughts?

**Comment by kellyselden** [comment]
I've thought about this, and I'm on board with it, just haven't got to it. It would be a new primitive in `ember-macro-helpers` that would also pass you the `getValue` function for your macro to do lazy resolving. The API would look something like this:

```js
import someOtherComputed from 'ember-macro-helpers/some-other-computed';

export default someOtherComputed(([condition, expr1, expr2], getValue) => {
  return getValue(condition) ? getValue(expr1) : getValue(expr2);
});
```

I'd love some help implementing this, as I've got the array bug on my table.

**Comment by JackCA** [comment]
ah, gotcha -- thanks for the info. It sounds like a tricky cross-library implementation. Maybe we can let this issue stay open to see if anyone else gets affected by this. I have an easy workaround in the meantime.

**Comment by kellyselden** [comment]
Here is an example of the progress so far https://github.com/kellyselden/ember-awesome-macros/compare/master...lazy-computed

**Comment by kellyselden** [comment]
Should be fixed now in v0.35.0
